Downstream cherry picks

assignedgod opened the pull request:
<!-- Guidelines: https://docs.spacestation14.io/en/getting-started/pr-guideline -->

## About the PR
<!-- What did you change? -->
<!-- If this is a code change, summarize at high level how your new code works. This makes it easier to review. -->
Brings in some changes from downstream, new hair and some new contravend items
## Why / Balance
<!-- Discuss how this would affect game balance or explain why it was changed. Link any relevant discussions or issues. -->
Benefits of having a downstream, get some nice easy to do cherry picks based off our codebase from time to time 
## Technical details
<!-- Summary of code changes for easier review. -->
Lots of changes, will need thorough review so nothing slips through the cracks
## How to test
<!-- Describe a procedure to test this feature, along with expected output/behavior. -->
Look at emag vendors, contravend inventories, and of course hair
## Media
<!-- Attach media if the PR makes ingame changes (clothing, items, features, etc).
Small fixes/refactors are exempt. Media may be used in SS14 progress reports with credit. -->

## Requirements
<!-- Confirm the following by placing an X in the brackets [X]: -->
- [x] I have read [CONTRIBUTING.md](https://github.com/new-frontiers-14/frontier-station-14/blob/master/CONTRIBUTING.md) and and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [x] I have added media to this PR or it does not require an ingame showcase.
- [x] I have reviewed the [Ship Submission Guidelines](https://frontierstation.wiki.gg/wiki/Ship_Submission_Guidelines) if relevant.
- [x] I confirm that the content in this PR is my own work, and/or is properly attributed to the original author(s).
<!-- You should understand that not following the above may get your PR closed at maintainer’s discretion -->

## Breaking changes
<!-- List any breaking changes, including namespaces, public class/method/field changes, prototype renames; and provide instructions for fixing them. -->

**Changelog**
<!-- Add a Changelog entry to make players aware of new features or changes that could affect gameplay.
Make sure to read the guidelines and take this Changelog template out of the comment block in order for it to show up.
Changelog must have a :cl: symbol, so the bot recognizes the changes and adds them to the game's changelog. -->
<!--
:cl:
- add: Added fun!
- remove: Removed fun!
- tweak: Changed fun!
- fix: Fixed fun!
-->

Cygnall commented:
> > Please consider before we restore the emag to how it was, in the interest of Wizden compatibility: We have a lot of systems that whenever we pull down from Wizden they have diverged enough they have to be patched. It may be worth using their version of systems just to reduce workload. I have no argument on this in regards to the function of emags, but it's just food for thought. A few things like this slipped in, in the big upmerge. Wizden versus NF's food systems come to mind as another example.
> 
> I would agree with you, however a few things- Emag contraband inventories still exist as a system within the code so theres no difficulty or cost to having them. If emags were changed upstream and that system removed? Yeah, definitely, but they still have function. The cost to maintain this is literally nothing so long as we keep our code commented. We can talk about "what-if" difficult to maintain things a lot, but this is not an instance of that.

Fair enough. This really comes down to if Wizden's version, which allows easier access to some contraband, or the old version, which gates items behind an emag, is more fun/ better for balance, then. I have no comment on that.

assignedgod commented:
> > > Please consider before we restore the emag to how it was, in the interest of Wizden compatibility: We have a lot of systems that whenever we pull down from Wizden they have diverged enough they have to be patched. It may be worth using their version of systems just to reduce workload. I have no argument on this in regards to the function of emags, but it's just food for thought. A few things like this slipped in, in the big upmerge. Wizden versus NF's food systems come to mind as another example.
> > 
> > I would agree with you, however a few things- Emag contraband inventories still exist as a system within the code so theres no difficulty or cost to having them. If emags were changed upstream and that system removed? Yeah, definitely, but they still have function. The cost to maintain this is literally nothing so long as we keep our code commented. We can talk about "what-if" difficult to maintain things a lot, but this is not an instance of that.
> 
> Fair enough. This really comes down to if Wizden's version, which allows easier access to some contraband, or the old version, which gates items behind an emag, is more fun/ better for balance, then. I have no comment on that.
What it comes down to is the fact that the emag should have use for contraband inventories on our fork since its use on lathes is minimally impactful since merclathes are widely available. Wizden removed contraband inventories from vendors as emagging a lathe is super powerful on wizden, compared to here where it is less so

assignedgod commented:
Yeah, been procrastinating on that. Basically it does a few things, adds some emag vendor inventories, a few implanters, a lot of new hairstyles from imp, it buffs the chemical synthesis kit by adding (originally four, I lowered it to three) vials of vestine rather than just one. Some changes I have to remake is removing the weird truncating of the ammo tab of the uplink into the weaponry tab. It added some magical books but I commented those out as I wanted to test balance of them first. It adds a cloak that lets you turn invisible - This is insanely strong, and the original drain made it so that a microreactor could beat it out via charge, I removed the ability to put a micro in the cloak and limited it to non self recharching batteries. 
I removed all the implanters and left them as admin spawn only items for now ahead of further balance discussions
